src/display/text_layer.js

```diff
@@ -513,15 +516,8 @@ var renderTextLayer = (function renderTextLayerClosure() {
 
     _render: function TextLayer_render(timeout) {
       var textItems = this._textContent.items;
-      var styles = this._textContent.styles;
-      var textDivs = this._textDivs;
-      var viewport = this._viewport;
-      var bounds = this._bounds;
-      var enhanceTextSelection = this._enhanceTextSelection;
-
       for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(textDivs, viewport, textItems[i], styles, bounds,
-                   enhanceTextSelection);
+        appendText(this, textItems[i]);
```

Comment:
Since you're passing in `textItems` separately here, I'm wondering if it wouldn't make sense to still pass in `styles` in a similar way too?
E.g. re-add `var styles = this._textContent.styles;` above and make this function call `appendText(this, textItems[i], styles);`?

The reason that I'm asking is that this would get rid of the, comparatively complex given the others, assignment `var style = task._textContent.styles[geom.fontName];` in `appendText` and use the previous `var style = styles[geom.fontName];` instead.